### PR TITLE
gtk3: Use tracker3 for search

### PR DESCRIPTION
**Pull request by jtojnar**
Upstream [recommends](https://discourse.gnome.org/t/gnome-3-38-uses-tracker-3-by-default-but-one-core-app-still-uses-tracker-2/4191) to build GTK 3 with Tracker 3 enabled. Without it, according to my tests, the search only affects the current directory.

But previously `tracker` depended on `gtk3` through `libsoup` → `sysprof` → `gtk3`, which only got resolved in https://github.com/NixOS/nixpkgs/pull/101528, and, what is more, even now through `wrapGAppsHook`. To resolve this, I created `wrapGAppsNoGuiHook`, which does not propagate dependencies needed for graphical programs. I tried to document it, when I noticed GdkPixbuf loaders are not mentioned so I added a docs section before I forget.

This is yet untested, will need to check what happens when user tries to search in GTK file dialogue on systems without Tracker D-Bus service.

### CODE REVIEW

File: pkgs/development/libraries/gtk/3.x.nix [hunk 1]
```diff
@@ -32,6 +32,8 @@
 , sassc
 , x11Support ? stdenv.isLinux
 , waylandSupport ? stdenv.isLinux
+, trackerSupport ? stdenv.isLinux
+, tracker
 , mesa
 , wayland
 , wayland-protocols
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtk/3.x.nix:39]:
```suggestion
, mesa
, wayland
, wayland-protocols
, trackerSupport ? stdenv.isLinux
, tracker
```

Probably nicer to keep the wayland things together.

Comment (jtojnar) [reply pkgs/development/libraries/gtk/3.x.nix:39]: Thanks, it should not actually even go to `propagatedBuildInputs`.

**Comment by worldofpeace** [comment]
> This is yet untested, will need to check what happens when user tries to search in GTK file dialogue on systems without Tracker D-Bus service.

Pantheon would be the perfect test bench for this :+1: (though they plan to use tracker in the future)

**Review by worldofpeace — APPROVED** [review]
Code looks good, needs testing though.

**Comment by worldofpeace** [comment]
I believe if we merge this to gnome-3.38 we'll still get an eval for it? (I think things shouldn't go to staging untested)

**Comment by jtojnar** [comment]
We need not only eval but also runtime testing, I would wait with this until next staging is merged.

**Comment by jtojnar** [comment]
Running gedit linked against this GTK on Ubuntu, the file chooser seems to work and even search does seem to show some recursive results. Though it does not seem to be complete and complains in the logs:

```
(gedit:82407): Tracker-CRITICAL **: 22:29:36.021: tracker_sparql_connection_close: assertion 'TRACKER_IS_SPARQL_CONNECTION (connection)' failed

(gedit:82407): Gtk-CRITICAL **: 22:29:36.021: Could not init tracker3 search engine: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Tracker3.Miner.Files was not provided by any .service files
```

**Comment by jtojnar** [comment]
Did not encounter any issues in NixOS VM either.

**Pull request merged**